specification/transport/serial/serial.tex

```diff
@@ -0,0 +1,231 @@
+and any similar communication links that allow exchange of unstructured byte streams.
+Cyphal/serial may also be used to store Cyphal frames in files.
+\textbf{%
+    As of this version, the Cyphal/serial specification remains experimental.
+    Breaking changes affecting wire compatibility are possible.
+}
+
+As Cyphal/serial is designed to operate over unstructured byte streams,
+it defines a custom framing protocol, custom frame header format, and a custom integrity checking mechanism.
+
+\begin{CyphalSimpleTable}{Cyphal/serial transport capabilities\label{table:transport_serial_capabilities}}{|l X l|}
+    Parameter & Value & References \\
+
+    Maximum node-ID value &
+    65534 (16 bits wide). &
+    \ref{sec:basic} \\
+
+    Transfer-ID mode &
+    Monotonic, 64 bits wide. &
+    \ref{sec:transport_transfer_id} \\
+
+    Number of transfer priority levels &
+    8 (no additional levels). &
+    \ref{sec:transport_transfer_priority} \\
+
+    Largest single-frame transfer payload &
+    Unlimited. &
+    \ref{sec:transport_transfer_payload} \\
+
+    Anonymous transfers &
+    Available. &
+    \ref{sec:transport_route_specifier} \\
+\end{CyphalSimpleTable}
+
+\subsection{Framing}
+
+Cyphal/serial uses the ``Consistent Overhead Byte Stuffing'' (COBS) encapsulation method\footnote{%
+    See ``Consistent Overhead Byte Stuffing'', Stuart Cheshire and Mary Baker.
+    The COBS overhead is 1 byte in every 254 bytes of encapsulated data, which is about 0.4\%.
+} with zero byte as the frame delimiter.
+Due to the nature of COBS, the frame delimiter will not appear in the frame payload.
+A frame delimiter may terminate a frame and/or indicate the start of a new frame.
+The number of frame delimiters between adjacent frames may exceed one.
+
+\begin{figure}[H]
+    \centering
+    $$
+    \texttt{\huge{...}}%
+    \underbrace{\texttt{\huge{0}}}_{\substack{\text{frame} \\ \text{delimiter}}}%
+    \underbrace{\texttt{\huge{<frame>}}}_{\substack{\text{COBS-encoded} \\ \text{frame contents}}}%
+    \underbrace{\texttt{\huge{0}}}_{\substack{\text{frame} \\ \text{delimiter}}}%
+    \underbrace{\texttt{\huge{<frame>}}}_{\substack{\text{COBS-encoded} \\ \text{frame contents}}}%
+    \underbrace{\texttt{\huge{0}}}_{\substack{\text{frame} \\ \text{delimiter}}}%
+    \texttt{\huge{...}}%
+    $$
+    \caption{COBS framing\label{fig:transport_serial_cobs}}
+\end{figure}
+
+A frame consists of two parts:
+the fixed-size header (section~\ref{sec:transport_serial_header})
+immediately followed by the payload (section~\ref{sec:transport_serial_payload}).
+The header contains a dedicated header-CRC field which allows implementations to detect frame corruption early.
+
+Neither the underlying medium nor the Cyphal/serial transport layer impose any restrictions on the maximum frame size,
+which allows all Cyphal/serial transfers to be single-frame transfers\footnote{%
+    Lack of multi-frame transfers simplifies implementations significantly.
```

Comment:
It is placed in the footnote, not in the main text. Footnotes are used for non-normative statements throughout the text along with the blue boxes. Changing this would require updating most of the text so I am inclined to keep it.

Reply:
It's still unquantified. I don't like have things like "it's great!" or "that would be bad" in our specification. Instead how about: "Omitting multi-frame transfers as a requirement is expected to simplify implementations."

Reply:
Updated.